### PR TITLE
Feedback modal

### PR DESCRIPTION
**Pull request by Solexi**
# Add Feedback Modal After Reset Button Click

_Adds a feedback modal that appears after the reset button is clicked in the DraftModal component._

## How should this be tested?

1. Click the reset button.
2. Verify that the feedback modal appears confirming the reset action.

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/3052

### CODE REVIEW

File: public/locales/en/wiki.json [hunk 1]
```diff
@@ -141,5 +141,7 @@
   "NewWikiDraft-1": "A draft has been loaded for your new wiki page. Would you like to reset the page back to a blank wiki?",
   "NewWikiDraft-2": "Note: Resetting will erase all current content and start with a clean slate",
   "ExistingWikiDraft-1": "Your content has been loaded from a saved draft. Would you like to reset it to the most recent wiki content instead?",
-  "ExistingWikiDraft-2": "Note: Resetting will overwrite any changes made since the last save."
+  "ExistingWikiDraft-2": "Note: Resetting will overwrite any changes made since the last save.",
+  "FeedbackModalHeader": "Wiki has been successfully reset!",
+  "FeedbackModalContent": "All previous content has been cleared, and the page is now refreshed with the new content."
```
Comment (kesar) [inline public/locales/en/wiki.json:146]: remember we need to get it in the other languages 👍🏻

**Comment by Tolu-Mals** [comment]
It helps if you include the figma link, so anyone can review against it

File: src/components/CreateWiki/EditorModals/DraftModal.tsx
Comment (Tolu-Mals) [inline src/components/CreateWiki/EditorModals/DraftModal.tsx:1]: These are two separate components, a draft modal and then a reset feedback modal. It's not ideal to put one component inside another component except they're also embedded like that visually(like one of them is literally inside the other one in the app), it helps with being able to find the components.

However, these components are related in the sense that they're both related to draft notifications, so you can move the reset feedback modal into the `useDraftNotifications` hook and return it from there just like the draft modal. The actual component itself should be just under the draft modal.

File: src/components/CreateWiki/EditorModals/DraftModal.tsx [hunk 2]
```diff
@@ -21,6 +24,38 @@ interface DraftModalProps {
...
+    setFeedbackModalOpen(true)
+
+    setTimeout(() => {
+      setFeedbackModalOpen(false)
+
+      router
+        .push(window.location.pathname, undefined, { shallow: true })
+        .then(() => {
+          window.location.reload()
+        })
```
Comment (Tolu-Mals) [inline src/components/CreateWiki/EditorModals/DraftModal.tsx:55]: We can simplify this logic. If you use a React hook to get the params, you can create a boolean variable `showResetFeedbackModal` using the param value, which will automatically get updated when the URL params changes. You can use this to conditionally render the modal in addition to an `isFeedbackModalOpen` state.  The `isFeedbackModalOpen` state will allow you stop showing the modal without having to do another page reload.

File: src/components/CreateWiki/EditorModals/DraftModal.tsx [hunk 3]
```diff
@@ -60,19 +95,19 @@ export const DraftModal = ({ isOpen, onClose, isNewWiki }: DraftModalProps) => {
...
               Reset
             </Button>
           </ModalBody>
         </ModalContent>
       </Modal>
+      <FeedbackModal
+        isOpen={isFeedbackModalOpen}
+        onClose={() => {
+          setFeedbackModalOpen(false)
+          router.push(`${window.location.pathname}`)
```
Comment (Tolu-Mals) [inline src/components/CreateWiki/EditorModals/DraftModal.tsx:108]: there's no need to remove url params after closing the modal

Comment (Tolu-Mals) [inline src/components/CreateWiki/EditorModals/DraftModal.tsx:108]: there's no need to remove url params after closing the modal

File: src/components/Layout/Navbar/NavMenu.tsx
Comment (Tolu-Mals) [inline src/components/Layout/Navbar/NavMenu.tsx:1]: For future prs, try to create your branch from the latest version of the main branch, ie. so we don't have changes like these from your previous prs show in the pr changes.

File: src/components/CreateWiki/EditorModals/DraftModal.tsx [hunk 4]
```diff
@@ -10,17 +10,33 @@ import {
   Button,
 } from '@chakra-ui/react'
 import { useTranslation } from 'next-i18next'
+import { useEffect, useState } from 'react'
 import { RiInformationLine } from 'react-icons/ri'
+import FeedbackModal from './FeedbackModal'
+import { useRouter } from 'next/router'
```
Comment (SidharthK2) [inline src/components/CreateWiki/EditorModals/DraftModal.tsx:16]:
```suggestion
import { RiInformationLine } from 'react-icons/ri'
```
Pls remove unused imports, also fyi linter only works on staged files in ep-ui

**Review by Tolu-Mals — APPROVED** [review]

**Review by Softdev1 — APPROVED** [review]
image
the page is not translated

**Pull request merged**